gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -410,6 +473,84 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
     return null;
   }
 
+  @VisibleForTesting
+  boolean isGoogleS2AEnabled() {
```

Comment:
qq, are there non-google S2A configurations that we intend to support in the future?

Perhaps `isS2AEnabled()` for consistency

Reply:
For now S2A is always going to be only used in 1P (Google) cases. This might change in the future, but there is nothing concrete yet that it will.

Naming was set to isGooglS2AEnabled to make clear that S2A is being use when 1P creds are available. A minor detail is that we also name the parallel function in Golang the same: https://github.com/googleapis/google-cloud-go/blob/main/auth/internal/transport/s2a.go#L113 . Shall we keep naming same for consistency with Go? I'm also fine changing to `isS2AEnabled`.

Reply:
IMO, I think `isS2AEnabled` is more future-proof given the possibility of non-Google use cases.

Unless 1P and 3P S2A implementations are vastly different and that we would end up having two separate flags like `isGoogleS2AEnabled` and `isNonGoogleS2AEnabled` flags.